jsonschema-core.xml

```diff
@@ -313,43 +313,70 @@
-                            apply one or more subschemas to a particular location
-                            in the instance, and combine or modify their results
+                            Keywords that are permitted in a schema,
+                            but that do not do anything by themselves,
+                            though their value may be read by or referred to by another keyword.
                         </dd>
-                        <dt>reserved locations</dt>
+                        <dt>Arguments</dt>
                         <dd>
-                            do not directly affect results, but reserve a place
-                            for a specific purpose to ensure interoperability
+                            Arguments are properties in a schema that are used by an adjacent keyword,
+                            and whose meaning may only be defined in the presence of that keyword.
                         </dd>
                     </dl>
                     <t>
-                        Keywords may fall into multiple categories, although applicators
-                        SHOULD only produce assertion results based on their subschemas'
-                        results.  They should not define additional constraints independent
-                        of their subschemas.
+                        Some properties may be classified into multiple categories.
+                        Some keywords are also arguments to other keywords,
+                        and applicator keywords are both assertions and annotations.
                     </t>
                     <t>
-                        Keywords which are properties within the same schema object are referred to as adjacent keywords.
+                        The categories that a keyword belongs to is defined per keyword.
+                        Additionally, there are naming conventions that impose limits or requirements on the categories that keywords belong to:
+                    </t>
+                    <ul>
+                        <li>
+                            A keyword that begins with "annotation-" is never a core keyword or assertion keyword, but may be an annotation or reserved keyword. These keywords MUST NOT affect a validation result.
+                            <cref>
+                                This naming is subject to change.
+                            </cref>
+                        </li>
+                        <li>
+                            A keyword that begins with "$" is a core keyword, unless defined otherwise.
+                            (For example, "$comment" behaves as a reserved keyword.)
+                        </li>
+                    </ul>
+                    <t>
+                        If a validator encounters a keyword whose assertion behavior is unknown (a keyword besides an "annotation-" keyword),
+                        then attempting to compute the validation result SHOULD raise an error.
```

Comment:
Instead of "raise an error", would it be better to say the result is "indeterminate"? (Has the "indeterminate" concept made it into the spec yet? It might need to be defined.) I often use "raise an error" as shorthand, but not all languages have a concept of raising an error and even then, it's not required that they use that mechanism to report the indeterminate result. But, I'm probably being overly pedantic.

Reply:
Yes, that's more specific too.

Reply:
What does "attempting to compute the validation result" mean? Does it mean "it is incorrect for the evaluator to compute a validation result from this keyword"?